assignment-2/src/view/SmartComponents/SmartAnswersList.js

```diff
@@ -0,0 +1,55 @@
+import React, {Component} from 'react';
+import questionModel from "../../model/questionModel";
+import AnswersList from "../DumbComponents/AnswersList";
+import sOUserModel from "../../model/sOUserModel";
+import answerModel from "../../model/answerModel";
+import answersListPresenter from "../../presenter/answersListPresenter";
+import sOUserPresenter from "../../presenter/sOUserPresenter";
+
+
+const mapModelStateToComponentState = (questionModel, answerModel, props) => ({
+    question: getQuestionById(questionModel.state.questions, props.match.params.id),
+    newAnswer: answerModel.state.newAnswer
+});
+
+function getQuestionById(questions, id) {
+    for (let i = 0; i < questions.length; i++) {
+        if (questions[i].id === parseInt(id))
+            return questions[i];
+    }
+}
+
+export default class SmartAnswersList extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = mapModelStateToComponentState(questionModel, answerModel, props);
+
+        this.listener = () =>
+            this.setState(mapModelStateToComponentState(questionModel, answerModel, props));
+        answerModel.addListener("change", this.listener);
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
```

Comment:
You did not use the `prevState` and `snapshot` parameters, so you can just omit them. JS has no concept of "method signature", this means that you can pass how many arguments you want and the function may declare a different number of arguments with no problems.